metals/src/main/scala/scala/meta/internal/metals/UserConfiguration.scala

```diff
@@ -574,3 +588,29 @@ object TestUserInterfaceKind {
   object CodeLenses extends TestUserInterfaceKind
   object TestExplorer extends TestUserInterfaceKind
 }
+
+sealed trait ShowInferredType {
+  def showAll: Boolean = this == ShowInferredType.On
+  def isDisabled: Boolean = this == ShowInferredType.Off
+  def showCases: Boolean = this == ShowInferredType.Minimal
+  def isNotDisabled: Boolean = !isDisabled
+}
+
+object ShowInferredType {
```

Comment:
Probably ideal scenario would be a json config, but that is much harder to model in the UI. Something like git lenses would be great where we would have a separate webview with the settings and turning them on/off would show the results immediately on a sample code. I think we could do that later as a totally different setting and then deprecate these.

On the other hand, the boolean or enum settings are quite easy to toggle on/off, which might be more complex to do in the scenario I am suggesting. We could just keep toogle options that would enable/disable a subset of configuration :thinking:

Reply:
> I think we could do that later as a totally different setting and then deprecate these.

ok, sounds reasonable

Reply:
Also I think adding a settings page for those sounds like a great project for anyone who knows typescript a bit better :wink:

Reply:
ok, now as we're on the same page, I'm closing PR